### PR TITLE
Removed travis and replaced with github actions for CI

### PR DESCRIPTION
**Pull request by matthewhegarty**
**Problem**

Previous CI processes stopped working recently.  Travis is undergoing changing and their free model [is questionable for the future](https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works).  A few commentators online suggest moving to Github actions.

Moving to GitHub actions was also suggested in #1200.

**Solution**

I have ported the CI over to github actions, which offers the same features and is obviously integrated well into Github.

Also:

- I have updated details of supported versions (removed reference to unsupported versions of Django)
- Updated the Code of Conduct using the latest version [here](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html)
- Removed `travis.yml` - no longer needed
- Makefile: Fixed deprecation warning due to call to `django-admin.py`
- Added changes from #1200

#### Deprecation warnings

Since adding #1200, we get some deprecation warnings which I have fixed:

- Removal of `safe_datetime` module
- Set `USE_TZ` to False
- Addition of `ChangeList` `search_help_text` kwarg in `admin.py`

**Acceptance Criteria**

- Ran `tox` locally 
- I have tested this in my forked copy.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,130 +0,0 @@
...
-      env: DJANGO="Django==2.0.*" IMPORT_EXPORT_TEST_TYPE=sqlite
-
-install:
-  - pip install -q $DJANGO
-  - pip install -r requirements/base.txt
-  - pip install -r requirements/test.txt
-  - pip install coveralls
-script:
-   - PYTHONPATH=".:tests:$PYTHONPATH" python -Wall -m coverage run --omit='setup.py' --source=. tests/manage.py test core --settings=
-   - if python -c 'import sys; sys.exit(1 - (sys.version_info >= (3, 6)))'; then isort --check-only import_export tests; fi
```
Comment (manelclos) [inline .travis.yml:128]: I can't find the isort test on the github workflow, not a showstopper though.

**Review by manelclos — APPROVED** [review]
Great job, thanks!!

File: .travis.yml (hunk 1 above)
Comment (matthewhegarty) [reply .travis.yml:128]: Hey Manel - thanks for reviewing.  I'll take a look into isort

Comment (matthewhegarty) [reply .travis.yml:128]: Have added isort checks

**Pull request merged**